### PR TITLE
Visitor can view all listed articles

### PR DESCRIPTION
**Pull request by PaulineBA**
[PT LINK](https://www.pivotaltracker.com/story/show/172888047)

User Story

```
As a Visitor
In order to choose an article to read
I would like to be able to see the articles listed on a page
```

## Changes proposed in this pull request:
* We have used class component as we were unsure about how to use the functional component. Is it still ok to use class component at this stage? We haven't refactored it, let us know if this is a must ;-)

## What I have learned working on this feature:
* How to center placeholder in a column with semantic

## Packages/Gems added
- Semantic-ui-react-css

## Screenshots [(Client)](https://www.pivotaltracker.com/story/show/172888047/comments/214527842)

[
Screen Shot 2020-05-22 at 9 50 41 AM
](url)

### CODE REVIEW

File: cypress/integration/visitorCanViewAllListedArticles.feature.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    cy.server();
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/v1/articles",
+      response: "fixture:articleList.json",
+    });
+    cy.visit("/");
+  });
+  it("header is shown", () => {
+    cy.get("#header");
```
Comment (erikbjoern) [inline cypress/integration/visitorCanViewAllListedArticles.feature.js]: Could contain .contains('Daily News Sense')

File: cypress/integration/visitorCanViewAllListedArticles.feature.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+describe("visitor can view all listed articles", () => {
+  beforeEach(() => {
+    cy.server();
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/v1/articles",
```
Comment (erikbjoern) [inline cypress/integration/visitorCanViewAllListedArticles.feature.js]: We don't have a v1 namespace actually! - remember to edit the axios base url also.

**Review by kermit-klein — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -6,9 +6,12 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
+    "axios": "^0.19.2",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
     "react-scripts": "3.4.1",
+    "semantic-ui-css": "^2.4.1",
+    "semantic-ui-react": "^0.88.2",
```
Comment (erikbjoern) [inline package.json:14]: cross-check this with what you wrote in PR overview (small thing...)

**Review by erikbjoern — CHANGES_REQUESTED** [review]
I left a few comments. Also - are you importing the stylesheet anywhere? Couldn't see it.

File: cypress/integration/visitorCanViewAllListedArticles.feature.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      response: "fixture:articleList.json",
```
Comment (oliverochman) [inline cypress/integration/visitorCanViewAllListedArticles.feature.js]: we use snake_case for json files

File: cypress/integration/visitorCanViewAllListedArticles.feature.js (hunk 1 above)
Comment (oliverochman) [inline cypress/integration/visitorCanViewAllListedArticles.feature.js]: out of scope for this feature

File: src/components/ArticleList.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import React, { Component } from "react";
+import axios from "axios";
+import { Placeholder, Grid } from "semantic-ui-react";
+
+export default class ArticleList extends Component {
```
Comment (oliverochman) [inline src/components/ArticleList.jsx]: we can have a session turning this component into a functional component with react hooks tomorrow

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: cypress/integration/visitorCanViewAllListedArticles.feature.js (hunk 2 above)
Comment (jysmys) [reply cypress/integration/visitorCanViewAllListedArticles.feature.js]: ok good to know, thanks!

File: cypress/integration/visitorCanViewAllListedArticles.feature.js (hunk 4 above)
Comment (kermit-klein) [reply cypress/integration/visitorCanViewAllListedArticles.feature.js]: this hasn't been solved yet

File: cypress/integration/visitorCanViewAllListedArticles.feature.js (hunk 2 above)
Comment (kermit-klein) [reply cypress/integration/visitorCanViewAllListedArticles.feature.js]: you forgot to change URL in index.js

File: src/components/Header.jsx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import React from "react";
 import { Grid } from "semantic-ui-react";
+import "../css/Header.css"
```
Comment (kermit-klein) [inline src/components/Header.jsx:3]: Are we going to have multiple css files ?
I think we shouldn't

File: src/index.js [hunk 7]
```diff
@@ -1,13 +1,12 @@
...
+import "semantic-ui-css/semantic.min.css";
+import axios from "axios";
 
-ReactDOM.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
-  document.getElementById('root')
-);
+axios.defaults.baseURL = "http://localhost:3000/api/v1";
```
Comment (erikbjoern) [inline src/index.js]: This url shouldn't contain /v1 either

**Review by erikbjoern — CHANGES_REQUESTED** [review]

**Pull request merged**